### PR TITLE
[#600] Add health metric

### LINKED ISSUE #600: Consider adding health metrics
Similar to https://github.com/nspcc-dev/neofs-node/pull/1614

### PR DESCRIPTION
**Pull request by KirillovDenis**
close #600 
Signed-off-by: Denis Kirillov <denis@nspcc.ru>

### CODE REVIEW

**Comment by alexvanin** [comment]
@realloc storage ode will produce these values as health metric:
```go
	HealthStatus_name = map[int32]string{
		0: "HEALTH_STATUS_UNDEFINED",
		1: "STARTING",
		2: "READY",
		3: "SHUTTING_DOWN",
	}
```

Gateways will be simpler for now: 1 - ready, 0 - starting. Just notifying.

**Review by alexvanin — DISMISSED** [review]

**Comment by KirillovDenis** [comment]
> @KirillovDenis also consider putting `cmd/s3-gw/app_healthy.go` into `cmd/s3-gw/app_metrics.go`.

I suppose we can do this in https://github.com/nspcc-dev/neofs-s3-gw/issues/591. It will require some refactoring anyway.

File: cmd/s3-gw/app_healthy.go [hunk 1]
```diff
@@ -1,46 +1,46 @@
...
+func newGateMetrics() *GateMetrics {
+	stateMetric := newStateMetrics()
+	stateMetric.register()
+
+	return &GateMetrics{
+		stateMetrics: stateMetric,
+	}
+}
+
+func newStateMetrics() stateMetrics {
```
Comment (masterSplinter01) [inline cmd/s3-gw/app_healthy.go:29]: I’m not sure but function starting with “new” and not returning a pointer — doesn’t it seem confusing?

Comment (masterSplinter01) [reply cmd/s3-gw/app_healthy.go:29]: Maybe we can drop the function? And move the content into ‘newGateMetrics’?

**Review by masterSplinter01 — APPROVED** [review]

**Pull request merged**